### ISSUE TITLE
Recursive type resolution for TypedDict

### ISSUE THREAD

**Issue by dumbPy** [microsoft/pyright#545]
**Is your feature request related to a problem? Please describe.**  
Not sure if this is a bug. Hence a feature request
```python
class Inner(TypedDict):
    inner_key: int

class Outer(TypedDict):
    outer_key: Inner

i: Inner = {"inner_key":1} #<-- works
o1: Outer = {"outer_key":{"inner_key":1}} #<-- throws an error
o2: Outer = {"outer_key": i} #<-- works
```
o1 throws an error,
```
Expression of type 'dict[str, dict[str, int]]' cannot be assigned to declared type 'Outer'
  'dict[str, dict[str, int]]' is incompatible with 'Outer'Pyright
```

**Describe the solution you'd like**
Would be great if pyright can resolve these, atleast for typeddict

**Additional context**
N.A.

**Comment by erictraut** [comment]
Thanks for the bug report.

You're correct. Pyright's bidirectional type inference mechanism (which uses the "expected" type to infer the type of a dictionary literal) contains special-case code for TypedDict, but this code works only one level deep. Making it work to an arbitrary depth will require significant work.

In the meantime, you can work around the issue by using a temporary variable with a type annotation.

**Comment by dumbPy** [comment]
Suppressing the error with `#type: ignore` also works perfectly

```python
    class Inner(TypedDict):
        inner_key: int

    class Outer(TypedDict):
        outer_key: Inner

    i: Inner = {"inner_key":1}
    o1: Outer = {"outer_key":{"inner_key":1}} #type: ignore
    o2: Outer = {"outer_key": i}

    i2 = o1["outer_key"] #<-- i2 inferred as Inner correctly
```

**Comment by erictraut** [comment]
Thanks again for the bug report. This will be fixed in the next published version of pyright.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.27, which I just published.
